app.py

```diff
@@ -15,9 +15,262 @@
+    """Extract error from command log."""
+    value = "Command exited with non-zero status code (1):"
+    if value in text:
+        text = text.replace(value, "")
+    text = " ".join(text.split())
+    return text
+
+
+def process_exit_status_error(text: str) -> str:
+    """Process and simplify error exit status."""
+    if "ERROR:" in text:
+        text = text.replace("ERROR: ", "")
+    if "Command errored out with exit status 1:" in text:
+        text = text.replace("Command errored out with exit status 1:", "exitStatus1")
+    if "Could not find a version that satisfies the requirement" in text:
+        text = text.replace("Could not find a version that satisfies the requirement", "CouldNotFindVersionRequirement")
+    if "Failed to successfully execute function" in text:
+        text = text.replace(
+            "Failed to successfully execute function in Python interpreter", "FailedToExcecuteFunction/Module"
+        )
+    if " (from versions:" in text:
+        text_arr = text.split(" (from versions:")
+        if len(text_arr) == 2:
+            text = text_arr[0]
+    return text
+
+
+def process_error_info(text: str) -> str:
+    """Process and simplify error info."""
+    if isinstance(text, str):
+        text = " ".join(text.split())
+        if "ERROR:" in text:
+            text = text.replace("ERROR: ", "")
+        if " [Errno 2] No such file or directory: " in text:
+            text = text.replace(" [Errno 2] No such file or directory: ", "DirectoryNotFound: ")
+        if "NameError:" in text:
+            text = text.replace(" name ", " ")
+            text = text.replace(" is not defined", "")
+        if "ModuleNotFoundError:" in text:
+            text = text.replace("No module named ", "")
+        if "Check the logs" in text:
+            text = text.replace("Check the logs for full command output.", "CheckTheLogs")
+        if "No matching distribution found" in text:
+            text = text.replace("No matching distribution found", "NoMatchingDistributionError")
+    return text
+
+
+def check_is_instance(text: str) -> str:
+    """Check if text is a string instance."""
+    if not isinstance(text, str):
+        text = ""
+    return text
+
+
+# Stopwords
+def remove_stopwords(text: str) -> str:
+    """Remove stopwords from text, stopwords set from nltk."""
+    text_arr = text.split()
+    words = [w for w in text_arr if w not in stopwords.words("english")]
+    return " ".join(words)
+
+
+def preprocess_document(solver_path: str) -> DataFrame:
+    """Preprocess document with all helperfunctions listed above to output dataframe with error_processed column."""
+    error_documents = []
+    for path in pathlib.Path(solver_path).iterdir():
+        if path.is_file():
+            document_id = str(path).split("/")[-1]
+            _LOGGER.debug("Processing solver document %r", document_id)
+            current_file = open(path, "r")
+            data = json.load(current_file)
```

Comment:
Please use context manager to properly handle file opening/closing:

```suggestion
            with open(path, "r") as current_file:
                data = json.load(current_file)
```